examples/chat/client/tasks.nim

```diff
@@ -81,25 +92,76 @@ proc statusContext*(arg: ContextArg) {.async, gcsafe, nimcall,
   resetContext()
 
 proc new(T: type UserMessageEvent, wakuMessage: WakuMessage): T =
-  let topic = wakuMessage.contentTopic
+  let
+    topic = wakuMessage.contentTopic
+    topicSplit = topic.split('/')
```

Comment:
From the perspective of someone who hasn't been coding this feature for some time, it would make it easier to digest if the `contentTopic` was parsed in to an object, and from there, each constituent part could be identified with a name. Something like this:
```nim
import
  std/[sequtils, strutils]

type
  Encoding* {.pure.} = enum
    Protobuffer* = "proto"
    RLP* = "rlp"
    
  ContentTopic* = ref object
    appName*: string
    appVersion*: int
    contentTopicName*: string
    encoding*: Encoding

  ParseError* = object of ExampleClientError # Maybe we should create this error type

proc parseContentTopic(contentTopic: string): ContentTopic {.raises: [Defect, ParseError].} =
  let topicSplit = contentTopic.split('/').keepIf(t => t != "") # remove empties
  
  if topicSplit.len != 4: raise ParseError fmt"'{contentTopic}' has an insufficient number of parts"
  
  var version: int
  try:
     version = topicSplit[1].parseInt
  except ValueError as e:
    raise ParseError fmt"Could not parse '{topicSplit[1]}' as a version integer"
  
  var encoding: Encoding
  try:
    encoding = parseEnum[Encoding](topicSplit[3])
  except ValueError as e:
    raise ParseError fmt"Could not parse '{topicSplit[3]}' as an encoding"

  return ContentTopic(
    appName: topicSplit[0]
    appVersion: version,
    contentTopicName: topicSplit[2],
    encoding: encoding
  )
```
And then the following would change:
```suggestion
var topic: ContentTopic
try:
  topic = wakuMessage.contentTopic.parseContentTopic
except ParseError as e:
  # do something with the error
  error "Failed to parse content topic, contentTopic=wakuMessage.contentTopic
```
Which could then be used like:
```nim
case fmt"{topic.appName}/{topic.appVersion}":
  of toyChat2App:
    if topic.encoding == Encoding.Protobuffer:
      # ...
```

Reply:
Thank you for writing this up! Excellent recommendations.

One of the tricky things is that content topic formatting is only a recommendation in the waku v2 specs. To my understanding, some apps using waku v2 (e.g. WalletConnect) use a different scheme for content topic strings.

On the other hand, as is already the case, the example client can be opinionated and enforce the recommended scheme. In which case having types and errors and a parser proc makes a a lot of sense — less fragile and more DRY, for sure.

Once we pull waku inside the `nim_status` library, the types, etc. you suggested should also be implemented within `nim_status` and offered as a re-export of `nim_status/client.nim`. The the example client's commands, tasks, etc. can make use of that facility.

If you're okay with it, let's hold off on implementing those helpers in this PR and do it as part of the refactor to pull waku inside `nim_status` (or a PR that follows on that refactor).